### ISSUE TITLE
Add IP and host of the registrant to the admin notification mail

### ISSUE THREAD

**Issue by AlfredSK** [friendica/friendica#7330]
### Expected behavior
Would be very helpful to have the IP and the host of the registrant in the registration notification mail the admin is receiving when register_policy is set to APPROVAL.

Something like this would be nice:

`You've received a registration request from 'Full Name (nickname)' at friendica.tld from 12.345.678.90 (host123.dom.example.com).`

Or:
```
You've received a registration request at friendica.tld.
Name: full name
Nickname: nickname
Host: 12.345.678.90 (host123.dom.example.com).
```

There are - unfortunately - registrations of spammers or people with other bad intentions. The IP and hostname can help to identify those unwanted registrations.

There is a PHP function for this:

` gethostbyaddr ( string $ip_address ) : string`

Side note: An additional option would be to make the 'note to admin' field mandatory.

### Actual behavior
At the moment the admin only gets the the full name used during registration.
`You've received a registration request from 'Full Name' at friendica.tld`

### Friendica version you encountered the problem
2019.09-dev

### Friendica source (git, zip)
git

### PHP version
7.2

### SQL version
10.1 MariaDB

**Comment by AlfredSK** [comment]
I have added the email address of the registering user to the list of desired information.

Reason: I got a registration lately and unfortunately clicked on 'deny' (by accident). I wanted to contact the user about that mishap and tell them to retry the registration. But I found out that there is no information about the new user's email address - not in Friendica and also not in the registration notification mail. I couldn't reach out to the user.

**Comment by tobiasd** [comment]
> Side note: An additional option would be to make the 'note to admin' field mandatory.

In theory it is. And the field requirements worked before it was made translate-able. But there are also browsers that ignore the parameter for input fields.

Just tested it, chromium respects the "required" field, so their used browser seems to ignore the request to the browser.

**Comment by AlfredSK** [comment]
Yep. I think this feature is already implemented (and seems to work from my experience). I should remove that from this (old) feature request.

**Comment by MrPetovan** [comment]
The field requirement shouldn't depend on the browser, this is a serious oversight in the form process.

**Comment by tobiasd** [comment]
Modern browsers work as expected. But there are browsers that wont work the way modern browsers do. IE6, elinks, lynx and so on. All webkit and gecko based browsers of the last decade will react the way intended I think. One could add a check in the backend as well though to catch scripts.

**Comment by MrPetovan** [comment]
This is what we should have done first not to have to rely on an HTML attribute.

**Comment by AlfredSK** [comment]
As interesting your conversation about that flaw in the registration page is 😊 I feel the urge to tell you that the protection against bots or scripted registration attempts works very well.

I didn't see a single successful registration request by a bot or script in the last years. Not because there are no attempts (I have a lot in my log) but because they all fail to fulfill all requirements.

**Comment by MrPetovan** [comment]
What the protection against bots and scripted registration again?

**Comment by AlfredSK** [comment]
- the full name needs at least one space character (optional)
- the email address needs to be valid and owned by the user (login details are sent there)
- the note for the admin is mandatory (must contain something)
- there is a hidden field - it's a trap! (must be empty)

**Comment by AlfredSK** [comment]
In case you are curious this is how it looks in Apache's log files:

**access.log:**
```
143.131.13.48 - - [04/Apr/2021:07:37:17 +0000] "GET /register HTTP/1.1" 200 21790 "-" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
143.131.13.48 - - [04/Apr/2021:07:37:25 +0000] "POST register HTTP/1.1" 400 3829 "https://libranet.de/register" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
37.210.62.63 - - [04/Apr/2021:07:45:57 +0000] "GET /register HTTP/1.1" 200 21790 "-" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
37.210.62.63 - - [04/Apr/2021:07:46:05 +0000] "POST register HTTP/1.1" 400 3829 "https://libranet.de/register" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
207.204.96.30 - - [04/Apr/2021:07:49:03 +0000] "GET /register HTTP/1.1" 200 21790 "-" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
207.204.96.30 - - [04/Apr/2021:07:49:09 +0000] "POST register HTTP/1.1" 400 3829 "https://libranet.de/register" "Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36"
```

**error.log:**
```
[Sun Apr 04 07:37:25.713324 2021] [core:error] [pid 3467377:tid 140057113511680] [client 143.131.13.48:6508] AH00126: Invalid URI in request POST register HTTP/1.1, referer: https://libranet.de/register
[Sun Apr 04 07:46:05.244892 2021] [core:error] [pid 3467377:tid 140057105118976] [client 37.210.62.63:2319] AH00126: Invalid URI in request POST register HTTP/1.1, referer: https://libranet.de/register
[Sun Apr 04 07:49:09.835901 2021] [core:error] [pid 3484543:tid 140056970901248] [client 207.204.96.30:64443] AH00126: Invalid URI in request POST register HTTP/1.1, referer: https://libranet.de/register
```

**Comment by MrPetovan** [comment]
Oooh, so manual browsers would fix complete the invalid URI `register` to the expected `/register`, but automated tools wouldn't? Interesting. These errors aren't even related to the protections you listed earlier. Thank you for the elaboration!

**Comment by AlfredSK** [comment]
Oh! Right. I didn't notice the missing `/`. :-)
